extensions/azure-keyvault/runtime/src/main/java/io/quarkiverse/azure/keyvault/secret/runtime/config/KeyVaultSecretConfigUtil.java

```diff
@@ -0,0 +1,85 @@
+package io.quarkiverse.azure.keyvault.secret.runtime.config;
+
+import com.azure.security.keyvault.secrets.models.KeyVaultSecretIdentifier;
+
+public class KeyVaultSecretConfigUtil {
+    private static final String AZURE_KEYVAULT_PREFIX = "kv//";
+    private static final String AZURE_KEYVAULT_ENDPOINT_PREFIX = "https://";
+    private static final String AZURE_VAULT_URL_FORMAT = "https://%s.vault.azure.net/secrets/%s/%s";
+
+    private KeyVaultSecretConfigUtil() {
+    }
+
+    /**
+     *
+     * @returns a {@link KeyvaultSecretIdentifier} from the specified input or <code>null</code> if the input does not start with {@link AZURE_KEYVAULT_PREFIX}.
+     *
+     * @throws IllegalArgumentException if the input cannot otherwise be parsed.
+     * 
+     */
+
+    static KeyVaultSecretIdentifier getSecretIdentifier(String input, String defaultEndpoint) {
+
+        if (!input.startsWith(AZURE_KEYVAULT_PREFIX)) {
+            return null;
+        }
+
+        String resourcePath = input.substring(AZURE_KEYVAULT_PREFIX.length());
+        String[] tokens = resourcePath.split("/");
+
+        String kvName = null;
```

Comment:
Let's think more carefully about the contract we are creating here. There are several ways the input can be considered invalid, but the result is not the same in all cases. Is that what we want? In the current implementation, the invalid input of not starting with `AZURE_KEYVAULT_PREFIX` causes a `null` return. However, other forms of syntactic invalidity cause an `IllegalArgumentException` to be thrown. Why?

It seems more consistent to return null in all cases unless the input is syntactically valid.

Reply:
I discussed this with @galiacheng . After discussion I think IAE is indeed appropriate.